src/Functions/DateTimeTransforms.h

```diff
@@ -57,25 +60,51 @@ struct ZeroTransform
     static UInt16 execute(UInt16, const DateLUTImpl &) { return 0; }
 };
 
+template <DateTimeOverflowMode date_time_overflow_mode = default_date_time_overflow_mode>
 struct ToDateImpl
 {
     static constexpr auto name = "toDate";
 
     static UInt16 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
     {
-        return static_cast<UInt16>(time_zone.toDayNum(t.whole));
+        return execute(t.whole, time_zone);
     }
+
     static UInt16 execute(Int64 t, const DateLUTImpl & time_zone)
     {
-        return UInt16(time_zone.toDayNum(t));
+        if constexpr (date_time_overflow_mode == DateTimeOverflowMode::SATURATE)
+        {
+            if (t < 0)
+                t = 0;
+            else if (t > 5662310399)    // 2149-06-06 23:59:59 UTC
```

Comment:
Can we introduce constants for that somewhere, or at least decompose the value into factors like `60 * 60 * 24 * 365 * 1249` (just making this up for the sake of an example)